File: kafka-connect-common/src/main/scala/com/datamountaineer/streamreactor/connect/sink/DbWriter.scala
package com.datamountaineer.streamreactor.connect.sink


import org.apache.kafka.connect.sink.SinkRecord


/**
  * Defines the contrsct for inserting a new Hbase row for the connect sink record
  */
trait DbWriter extends AutoCloseable {
  def write(records: Seq[SinkRecord]): Unit
}

File: kafka-connect-common/src/main/java/com/datamountaineer/streamreactor/connect/sink/DbWriter.java
package com.datamountaineer.streamreactor.connect.sink;

import org.apache.kafka.connect.sink.SinkRecord;

import java.util.Collection;

/**
 * Defines the contract for inserting a new entry fromthe connect sink record
 */
public interface DbWriter extends AutoCloseable {
    void write(Collection<SinkRecord> records);
}
